packages/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol

```diff
@@ -480,16 +478,16 @@ interface ISuperfluid {
         //
         // App context
         //
-        // app allowance granted
-        uint256 appAllowanceGranted;
-        // app allowance wanted by the app callback
-        uint256 appAllowanceWanted;
-        // app allowance used, allowing negative values over a callback session
-        int256 appAllowanceUsed;
+        // app credit granted
+        uint256 appCreditGranted;
+        // app credit wanted by the app callback
+        uint256 appCreditWanted;
```

Comment:
mark it as appCreditWantedDeprecated and remove the usage of it throughout.

Reply:
make some additional comments re decodeCtx that it has abi compatibility requirement.